### PR TITLE
Add references and tutorials for `forge clone`

### PR DESCRIPTION
**Pull request by Troublor**
Now that `forge clone` command has been merged into `foundry` main repository in https://github.com/foundry-rs/foundry/pull/7576, here we also add command references, tutorials, and examples for `forge clone` in the foundry book. 

Credits to @ZhangZhuoSJTU and @Troublor.

### CODE REVIEW

File: src/projects/clone-a-verified-contract.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+## Clone a Verified Contract on Chain
+
+To clone an on-chain verified contract as a Forge project, use [`forge clone`](../reference/forge/forge-clone.md), say [WETH9](https://etherscan.io/address/0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2) on Ethereum mainnet:
+
+```sh
+{{#include ../output/clone_contract/forge-clone:command}}
+```
+
+This creates a new directory `WETH9`, configures it as a foundry project and clone all the source code of the contract into it. This also initializes a new `git` repository.
```
Comment (Evalir) [inline src/projects/clone-a-verified-contract.md:9]:
```suggestion
This creates a new directory `WETH9`, configures it as a foundry project and clones all the source code of the contract into it. This also initializes a new `git` repository.
```

File: src/projects/clone-a-verified-contract.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+```sh
+{{#include ../output/clone_contract/forge-clone:output}}
+```
+
+The cloned Forge project comes with an additional `.clone.meta` metadata file besides those ordinary files that a normal Forge project has.
+
+Let's check what the `.clone.meta` file looks like:
```
Comment (Evalir) [inline src/projects/clone-a-verified-contract.md:17]:
```suggestion
Let's see what the `.clone.meta` file looks like:
```

File: src/reference/forge/forge-clone.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+forge-clone - Clone a on-chain verified contract as a Forge project.
+
+### SYNOPSIS
+
+``forge clone`` [*options*] *address* [*root*]
+
+### DESCRIPTION
+
+Create a new Forge project by cloning the source code of an on-chain verified contract in the directory *root* (by default the current working directory).
```
Comment (Evalir) [inline src/reference/forge/forge-clone.md:13]:
```suggestion
Creates a new Forge project by cloning the source code of an on-chain verified contract in the directory *root* (by default the current working directory).
```

File: src/reference/forge/forge-clone.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+By default, `forge clone` clones contracts from Ethereum mainnet (via Etherscan), but it is also possible to clone from other EVM-compatible blockchains that foundry supports, e.g., BNB Smart Chain (BSC), by specifying chain ID: `--chain <ChainID>`.
+
+Obtaining data from Etherscan is subject to rate limit. `forge clone` requires two API calls to Etherscan to collect source code and deployment information, respectively. By default, `forge clone` will wait for *5* seconds between the two Etherscan invocations to avoid rate limit errors. 
+
+Specifying Etherscan API key via `--etherscan-api-key <API_KEY>` will increase Etherscan API rate limit and avoid the *5-second* wait time in `forge clone`.
+
+Just as `forge init`, `forge forge` will by default initialize a new git repository, install some submodules and create an initial commit message.
```
Comment (Evalir) [inline src/reference/forge/forge-clone.md:21]:
```suggestion
Just as `forge init`, `forge clone` will by default initialize a new git repository, install some submodules and create an initial commit message.
```

File: src/reference/forge/forge-clone.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+3. Clone a contract from BNB Smart Chain (BSC):
+    ```sh
+    forge clone --chain bsc 0x7862D9B4bE2156B15d54F41ee4EDE2d5b0b455e4 UniswapV3Pool 
+    ```
+
+### Metadata
+
+The cloned Forge project comes with an additional `.clone.meta` metadata file in the root directory.
+`clone.meta` is a compact JSON data file that contains the information of the on-chain contract instance. 
+The metadata include:
```
Comment (Evalir) [inline src/reference/forge/forge-clone.md:77]:
```suggestion
The metadata includes:
```

**Review by Evalir — APPROVED** [review]
Thanks! This looks great 🙏

**Comment by Troublor** [comment]
@all-contributors 
please add @Troublor for code.
please add @ZhangZhuoSJTU for code.

**Comment by DaniPopes** [comment]
@all-contributors please add @ZhangZhuoSJTU for code

**Comment by DaniPopes** [comment]
@all-contributors please add @ZhangZhuoSJTU for code

**Pull request merged**